### ISSUE TITLE
Endless build loop (again)

### ISSUE THREAD

**Issue by laeubi** [eclipse-m2e/m2e-core#1745]
With   `M2E - Maven Integration for Eclipse	2.6.1.20240304-0744	org.eclipse.m2e.feature.feature.group	Eclipse.org - m2e` I now encounter again an endless build loop with my tycho workspace, that is after a maven build is triggered, the same build runs again, triggering another build and so on... in contrast to before this is not one never ending build job but the job seem to complete but gets triggered again.

**Comment by laeubi** [comment]
It seems importing maven-surefire-plugin sources also trigger this.

**Comment by mickaelistria** [comment]
Does build triggers a change (probably via a mojo) in some visible project file that is not marked as derived in the workspace?

**Comment by alerosmile** [comment]
For me, that started when `org.eclipse.m2e.mavenarchiver` started to generate manifests including the current timestamp. The new timestamp causes a change in the model and therefore triggers a new build.

**Comment by HannesWell** [comment]
> For me, that started when org.eclipse.m2e.mavenarchiver started to generate manifests including the current timestamp.

Can you tell exactly when this started to happen?

**Comment by alerosmile** [comment]
No, not exactly. I have a workspace with many projects for quite a few years now. Those ‚endless‘ builds started anywhere by the end of 2022, I believe. The `org.eclipse.m2e.mavenarchiver` got introduced at that time.

I‘ve spent some time debugging m2e to find the cause for the build trigger. 

Now I‘ve added a profile which is active when inside Eclipse and this profile sets the timestamp to a static value. This solved that issue for me.

**Comment by Madjosz** [comment]
> Now I‘ve added a profile which is active when inside Eclipse and this profile sets the timestamp to a static value. This solved that issue for me.

@alerosmile Could you please give a more detailed explanantion what and how you did here and which timestamp you set?

**Comment by alerosmile** [comment]
@Madjosz My `maven-jar-plugin` configuration looks like this:
```
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-jar-plugin</artifactId>
  <configuration>
    <archive>
      ...
      <manifestEntries>
        <Built-Date>${project.build.outputTimestamp}</Built-Date>
        ...
      </manifestEntries>
    </archive>
  </configuration>
</plugin>
```

And that's how my profile looks like. It will automatically be activated in Eclipse:

```
<profile>
  <id>eclipse.m2e.target</id>
  <activation>
    <property>
      <name>m2e.version</name>
    </property>
  </activation>
  <build>
    <plugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-jar-plugin</artifactId>
        <configuration>
          <archive>
            <manifestEntries>
              <!-- Set this to a fixed value to prevent infinite Maven updates in m2e! -->
              <!-- An update changes this value which triggers an update which changes this value and so on... -->
              <Built-Date>--</Built-Date>
            </manifestEntries>
          </archive>
        </configuration>
      </plugin>
    </plugins>
  </build>
</profile>
```
